mlflow/pyfunc/__init__.py

```diff
@@ -1138,7 +1138,15 @@ def _load_model_or_server(
     try:
         client.wait_server_ready(timeout=90, scoring_server_proc=scoring_server_proc)
     except Exception as e:
-        raise MlflowException("MLflow model server failed to launch.") from e
+        if scoring_server_proc.poll() is None:
+            # the scoring server is still running but client can't connect to it.
+            # kill the server.
+            scoring_server_proc.kill()
+        server_output, _ = scoring_server_proc.communicate()
```

Comment:
Is `server_output` `str` or `bytes`?

Reply:
We have `text=True` param for `subprocess.Popen`, so it is `str`

Reply:
sure